### ISSUE TITLE
Bug in read_meta

### ISSUE THREAD

**Issue by thk686** [r-lib/pkgdown#594]
It has
```
yaml <- yaml::yaml.load_file(path)
```
at ```yaml``` is assumed to be a list but ```yaml.load_file``` can return a non-list.

Here's what I get:

```
> pkgdown::build_site()
Error: is.list(x) is not TRUE
> traceback()
6: stop(msg, call. = FALSE, domain = NA)
5: stopifnot(is.list(x), is.list(val))
4: utils::modifyList(meta, override)
3: as_pkgdown(pkg, override = override)
2: section_init(pkg, depth = 0, override = override)
1: pkgdown::build_site()
```

**Comment by thk686** [comment]
Turns out my yml file was there but empty. Not sure why. Perhaps I killed a site_build and it got clobbered.

**Comment by hadley** [comment]
Fixed by #598
